### PR TITLE
Add docker docs regarding file/logging permissions

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: Add docker docs regarding file/logging permissions
Additional Description: Explains default container logging and requirements to log to file
Risk Level: low
Testing:
Docs Changes: `docs/root/start/start.rst`
Release Notes: n/a
[Optional Runtime guard:]
[Optional Fixes #Issue] Touch #11836 
[Optional Deprecated:]

### CODE REVIEW

File: docs/root/start/start.rst [hunk 1]
```diff
@@ -160,6 +160,23 @@ for controlling access to an ``envoy`` socket from outside of the container.
...
+The ``envoy`` image sends application logs to ``/dev/stdout`` and ``/dev/stderr`` by default, and these
+can be viewed in the container log.
+
+If you send application, admin or access logs to a file output, the ``envoy`` user will require the
+necessary permissions to write to this file. This can be achieved by setting the ``ENVOY_UID`` and/or
+by making the file writeable by the envoy user.
+
+For example, to mount a log folder from the host and make it writable, you can:
+
+  $ mkdir logs
```
Comment (ggreenway) [inline docs/root/start/start.rst:172]: These lines are rendering poorly. I think you need to add `.. substitution-code-block:: none` above the 3 lines. There's an example of that earlier in the file.

Comment (phlax) [reply docs/root/start/start.rst:174]: added

**Comment by ggreenway** [comment]
In the future, please don't force-push to PRs; it makes them difficult to review. A merge into the PR branch is fine because it will be squashed when the PR is merged.

**Comment by phlax** [comment]
@ggreenway the problem with not force-pushing is that the PR needs to be rebased to master to be landable (i think)

**Comment by ggreenway** [comment]
> @ggreenway the problem with not force-pushing is that the PR needs to be rebased to master to be landable (i think)

No, you can just merge master into your branch if there are merge conflicts. No need to rebase.

**Pull request merged**